url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/math/response/Response.java

```diff
@@ -0,0 +1,669 @@
+			// to close distance, either on one or both sides accordingly
+			if (incentive.isPenalty()) {
+
+				for (int i = startIndex; i < endIndex; i++) {
+
+					temp = incentive.getBase() * values[i] / incentive.getPrice();
+					// System.out.println("Temp = " + temp);
+					overDiffTemp = (values[i] - temp) * awareness * sensitivity;
+					overDiff += overDiffTemp;
+					values[i] -= overDiffTemp;
+
+					// overDiff += values[i] - temp;
+					// values[i] = temp;
+
+				}
+				// System.out.println("Over Difference = " + overDiff);
+				double additive = overDiff / Constants.SHIFTING_WINDOW_IN_MINUTES;
+
+				switch (type) {
+
+				case "Both":
+
+					side = Constants.SHIFTING_WINDOW_IN_MINUTES / 2;
+
+					for (int i = 0; i < side; i++) {
+
+						int before = startIndex - i;
+						if (before < 0)
+							before += Constants.MINUTES_PER_DAY;
+						int after = endIndex + i;
+						if (after > Constants.MINUTES_PER_DAY - 1)
+							after %= Constants.MINUTES_PER_DAY;
+
+						values[before] += additive;
+						values[after] += additive;
+
+					}
+					break;
+
+				case "Left":
+
+					side = Constants.SHIFTING_WINDOW_IN_MINUTES;
+
+					for (int i = 0; i < side; i++) {
+
+						int before = startIndex - i;
+						if (before < 0)
+							before += Constants.MINUTES_PER_DAY;
+						values[before] += additive;
+
+					}
+					break;
+
+				case "Right":
+
+					side = Constants.SHIFTING_WINDOW_IN_MINUTES;
+
+					for (int i = 0; i < side; i++) {
+
+						int after = endIndex + i;
+						if (after > Constants.MINUTES_PER_DAY - 1)
+							after %= Constants.MINUTES_PER_DAY;
+
+						values[after] += additive;
+					}
+				}
+			}
+			// In case of reward a percentage of the close distances are moved in the
+			// window, either from one or both sides accordingly.
+			else {
+				side = Constants.SHIFTING_WINDOW_IN_MINUTES * 2;
+				switch (type) {
+
+				case "Both":
+
+					for (int i = startIndex - side; i < startIndex; i++) {
+
+						int index = i;
+
+						if (index < 0)
+							index += Constants.MINUTES_PER_DAY;
+
+						temp = incentive.getPrice() * values[index] / incentive.getBase();
+						overDiffTemp = (values[index] - temp) * awareness * sensitivity;
+						overDiff += overDiffTemp;
+						values[index] -= overDiffTemp;
+
+						// overDiff += values[index] - temp;
+						// values[index] = temp;
+					}
+
+					for (int i = endIndex; i < endIndex + side; i++) {
+
+						int index = i;
+
+						if (index > Constants.MINUTES_PER_DAY - 1)
+							index %= Constants.MINUTES_PER_DAY;
+
+						temp = incentive.getPrice() * values[index] / incentive.getBase();
+						overDiffTemp = (values[index] - temp) * awareness * sensitivity;
+						overDiff += overDiffTemp;
+						values[index] -= overDiffTemp;
+
+						// temp = incentive.getPrice() * values[index] /
+						// incentive.getBase();
+						// overDiff += values[index] - temp;
+						// values[index] = temp;
+					}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 263:279, 305:321.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.